common/commands/src/coconut/issue_credentials.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022-2023 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use clap::Parser;
+use nym_issue_credential::utils;
+
+use crate::context::SigningClientWithNyxd;
+use nym_network_defaults::NymNetworkDetails;
+use nym_validator_client::nyxd::Coin;
+
+#[derive(Debug, Parser)]
+pub struct Args {
+    /// Home directory of the client that is supposed to use the credential.
+    #[clap(long)]
+    pub(crate) client_home_directory: std::path::PathBuf,
+
+    /// A mnemonic for the account that buys the credential
+    #[clap(long)]
+    pub(crate) mnemonic: Option<bip39::Mnemonic>,
+
+    /// The amount of utokens the credential will hold. If recovery mode is enabled, this value
+    /// is not needed
+    #[clap(long, default_value = "0")]
+    pub(crate) amount: u64,
+
+    /// Path to a directory used to store recovery files for unconsumed deposits
+    #[clap(long)]
+    pub(crate) recovery_dir: std::path::PathBuf,
+}
+
+pub async fn execute(args: Args, client: SigningClientWithNyxd) {
+    let network_details = NymNetworkDetails::new_from_env();
+    let coin = Coin::new(
```

Comment:
kinda relating to the above. if you're going to be using the recovery mode, do you need to be constructing the coin?